xbmc/video/dialogs/GUIDialogVideoInfo.cpp

```diff
@@ -1664,6 +1664,15 @@ bool CGUIDialogVideoInfo::GetSetForMovie(const CFileItem *movieItem, CFileItemPt
     CFileItemPtr keepItem(new CFileItem(strKeep));
     keepItem->GetVideoInfoTag()->m_iDbId = currentSetId;
     listItems.AddFront(keepItem, 1);
+    // remove duplicate entry
```

Comment:
If you would move this logic up to before the clear item and the current set are added the code could be cleaner without having to remember to start at `listIndex = 2`.